src/libraries/System.Private.CoreLib/src/System/Nullable.cs

```diff
@@ -119,7 +122,18 @@ public static class Nullable
             if (nullableType.IsGenericType && !nullableType.IsGenericTypeDefinition)
             {
                 // instantiated generic type only
-                Type genericType = nullableType.GetGenericTypeDefinition();
+                Type genericType;
+                if (skipGenericDefinitionCache)
+                {
+                    RuntimeTypeHandle nativeHandle = nullableType.TypeHandle;
+                    genericType = null!;
+                    RuntimeTypeHandle.GetGenericTypeDefinition(new QCallTypeHandle(ref nativeHandle), ObjectHandleOnStack.Create(ref genericType));
```

Comment:
This won't build on Mono. I would leave this file alone, and instead changed the IsNullable check during `RuntimeTypeCache` initialization.

Reply:
@jkotas Fixed, please check.